zc_install/includes/nginx_conf/ngx_conf_server.txt

```diff
@@ -0,0 +1,320 @@
+		fastcgi_param	REQUEST_METHOD		$request_method;
+		fastcgi_param	CONTENT_TYPE		$content_type;
+		fastcgi_param	CONTENT_LENGTH		$content_length;
+		fastcgi_param	SCRIPT_NAME		$fastcgi_script_name;
+		fastcgi_param	REQUEST_URI		$request_uri;
+		fastcgi_param	DOCUMENT_URI		$document_uri;
+		fastcgi_param	DOCUMENT_ROOT		$document_root;
+		fastcgi_param	SERVER_PROTOCOL		$server_protocol;
+		fastcgi_param	GATEWAY_INTERFACE	CGI/1.1;
+		fastcgi_param	SERVER_SOFTWARE		nginx/$nginx_version;
+		fastcgi_param	REMOTE_ADDR		$remote_addr;
+		fastcgi_param	REMOTE_PORT		$remote_port;
+		fastcgi_param	SERVER_ADDR		$server_addr;
+		fastcgi_param	SERVER_PORT		$server_port;
+		fastcgi_param	SERVER_NAME		$server_name;
+		fastcgi_param	REDIRECT_STATUS		200;
+		
+		fastcgi_hide_header X-Powered-By;
+	}
+
+
+	#############################################################################################
+	# ZEN CART Pretty URLs Handler
+	#############################################################################################	
+	## The following block handles Pretty URLs
+	## Insert directives below as required
+	#
+	#
+	#
+	#
+
+
+
+	#############################################################################################
+	# ZEN CART 'zc_install' Folder
+	#############################################################################################	
+	## Allow all requests to specific static file types in 'zc_install'
+	location ~ %%store_folder%%/zc_install/(.+)\.(ico|js|css|jpg|gif|png|html)$ {
+		expires $zencart_expires;		
+	}
+	## Allow and fastrack internal requests to PHP files in subdirectories under 'zc_install'
+	location ~ %%store_folder%%/zc_install/(.+)/(.+)\.php$ {
+		internal;
+		error_page 418 = @zencart_php_handler;
+		return 418;
+	}
+	## Allow all requests to index.php file directly under 'zc_install'
+	location ~ %%store_folder%%/zc_install/index\.php$ {
+		error_page 418 = @zencart_php_handler;
+		return 418;
+	}
+	## Allow all requests to PHP files with names that start with 'ajax' directly under 'zc_install' 
+	location ~ %%store_folder%%/zc_install/ajax(.+)\.php$ {
+		error_page 418 = @zencart_php_handler;
+		return 418;
+	}
+	## Allow internal requests to 'zc_install/sql'
+	location ~ %%store_folder%%/zc_install/sql(.*)$ {
+		internal;
+	}
+	## Redirect all requests to 'zc_install' root to 'zc_install/index.php’
+	location ~ %%store_folder%%/zc_install(/?)$ {
+		return $scheme://$host%%store_folder%%/zc_install/index.php;
+	}
+	## Deny any other request to 'zc_install' and any subfolders
+	location ~ %%store_folder%%/zc_install {
+		return 403;
+	}
+
+
+	#############################################################################################
+	# ZEN CART 'includes' Folder
+	#############################################################################################	
+	## Allow and fastrack internal requests to 'includes/classes'
+	location ^~ %%store_folder%%/includes/classes {
+		internal;
+	}
```

Comment:
Why the need to be explicit about this `classes` folder? 

Why `classes` and nothing else? Why `classes` at all?